### PR TITLE
Advi docstr etc.

### PR DESCRIPTION
**Pull request by taku-y**
#775
- Add check for discrete variables. 
- Add logsumexp. 
- Add docstring. 

Could you see that code for discrete variable check is OK?

### CODE REVIEW

File: pymc3/variational/advi.py [hunk 1]
```diff
@@ -19,8 +22,50 @@
...
 ADVIFit = namedtuple('ADVIFit', 'means, stds, elbo_vals')
 
+def is_discreteRV(var):
+    dist = var.distribution
+
+    # Transformed distributions are continuous. (is it true?)
+    if isinstance(dist, TransformedDistribution):
+        return False
+
+    return isinstance(dist, Discrete)
```
Comment (twiecki) [inline pymc3/variational/advi.py:32]: Here's an example of a discrete check: https://github.com/pymc-devs/pymc3/blob/9b4ee10433cf5e92a631033a71d79334b503744d/pymc3/step_methods/metropolis.py#L94 Yours might be fine too, though.

**Comment by twiecki** [comment]
This is great! See my one question but otherwise we can merge.

File: pymc3/variational/advi.py (hunk 1 above)
Comment (twiecki) [inline pymc3/variational/advi.py:32]: Actually I think your check should be fine.

Comment (taku-y) [inline pymc3/variational/advi.py:32]: Thanks for explaining me how to check discrete variables.

**Pull request merged**